EPCPortalTesting/DataHandlerModelTests.cs

```diff
@@ -0,0 +1,65 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using EPCPortalWeb;
+using EPCPortalWeb.Models;
+using System.Collections.Generic;
+
+namespace EPCPortalTesting
+{
+    [TestClass]
+    public class DataHandlerModelTests
+    {
+        DataHandlerModel dataHandlerModel = new DataHandlerModel();
+
+        List<string> MockAPIReturn = new List<string>()
+            {
+                "1 fake street",
+                "2 fake street",
+                "3 fake street",
+                "4 fake street",
+                "5 fake street"
+            };
+
+        List<string> GetListOfAddressesByPostcode(string postcode)
+        {
+            List<string> ListOfAddresses = new List<string>();
+            //API Call
+            ListOfAddresses = MockAPIReturn;
+            return ListOfAddresses;
+        }
+
+        [TestMethod]
+        public void SearchForAddresses_WhenPostcodeIsSubmitted()
+        {
+            List<string> addresses = GetListOfAddressesByPostcode("RM96BF");
+            Assert.IsNotNull(addresses);
```

Comment:
Not sure of the goal behind this test. The method being called is part of the Test Class itself, rather than the DataHandlerModel that is focus of this fixture.

Reply:
Since the API was not set up at the time this was written i created the "Dummy COde" as a copy of what will be written into the class once the data was set up. this is a mock right now

Reply:
Understood, but goal of this test class is to unit test the DataHandlerModel, which you have done in the other tests within this class. This test doesn't fit with that particular goal. A separate class of integration tests could be used to make sure the Model is working with the live API caller as expected.

Also, in terms of mocking out the caller, I recommend using something like NSubstitute - I can demo that if you haven't used it before.